Crashlytics/Crashlytics/Components/FIRCLSProcess.c

```diff
@@ -524,19 +527,24 @@ void FIRCLSProcessRecordThreadNames(FIRCLSProcess *process, FIRCLSFile *file) {
     FIRCLSProcessGetThreadName(process, thread, name, THREAD_NAME_BUFFER_SIZE);
 
     FIRCLSFileWriteArrayEntryString(file, name);
+    FIRCLSFileWriteArrayEntryString(metricKitFile, name);
   }
 
   FIRCLSFileWriteArrayEnd(file);
   FIRCLSFileWriteSectionEnd(file);
+  FIRCLSFileWriteArrayEnd(metricKitFile);
+  FIRCLSFileWriteSectionEnd(metricKitFile);
 }
 
-void FIRCLSProcessRecordDispatchQueueNames(FIRCLSProcess *process, FIRCLSFile *file) {
+void FIRCLSProcessRecordDispatchQueueNames(FIRCLSProcess *process, FIRCLSFile *file, FIRCLSFile *metricKitFile) {
   uint32_t threadCount;
   uint32_t i;
 
   FIRCLSFileWriteSectionStart(file, "dispatch_queue_names");
+  FIRCLSFileWriteSectionStart(metricKitFile, "dispatch_queue_names");
```

Comment:
So thinking more about this, I feel like we should skip the part about writing thread names and not cross streams like this. If MetricKit doesn't provide it then we should just go with what they provide. If we want to do this, wait for customer feedback or solve it later. I feel like this is unintuitive implementation-wise, and seems like issues could come up with timing or missing MetricKit reports (eg. if we require thread names to process them successfully, and there isn't a Mach Exception signal triggered).